src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,177 @@
+    case WFindOnWebMenu::TrackSearchProperties::Artist:
+        return QObject::tr("Artist");
+    case WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle:
+        return QObject::tr("Title + Artist");
+    case WFindOnWebMenu::TrackSearchProperties::ArtistAndAlbum:
+        return QObject::tr("Album + Artist");
+    default:
+        return QObject::tr("Album");
+    }
+}
+
+const QString kSearchUrlSoundCloudArtist = QStringLiteral("https://soundcloud.com/search/people?");
+
+const QString kSearchUrlSoundCloudTitle = QStringLiteral("https://soundcloud.com/search/sounds?");
+
+const QString kSearchUrlSoundCloudAlbum = QStringLiteral("https://soundcloud.com/search/albums?");
+
+const QString kSearchUrlDiscogsGen = QStringLiteral("https://www.discogs.com/search/?");
+
+const QString kSearchUrlLastFmArtist = QStringLiteral("https://www.last.fm/search/artists?");
+
+const QString kSearchUrlLastFmTitle = QStringLiteral("https://www.last.fm/search/tracks?");
+
+const QString kSearchUrlLastFmAlbum = QStringLiteral("https://www.last.fm/search/albums?");
+
+const QString kSearchUrlDefault = QStringLiteral("https://soundcloud.com/search?");
+
+QString getServiceUrl(WFindOnWebMenu::Service service,
+        WFindOnWebMenu::TrackSearchProperties trackSearchProperties) {
+    switch (service) {
+    case WFindOnWebMenu::Service::Discogs:
+        return kSearchUrlDiscogsGen;
+    case WFindOnWebMenu::Service::LastFm:
+        if (trackSearchProperties == WFindOnWebMenu::TrackSearchProperties::Title ||
+                trackSearchProperties == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+            return kSearchUrlLastFmTitle;
+        } else if (trackSearchProperties == WFindOnWebMenu::TrackSearchProperties::Artist) {
+            return kSearchUrlLastFmArtist;
+        } else {
+            return kSearchUrlLastFmAlbum;
+        }
+    case WFindOnWebMenu::Service::SoundCloud:
+        if (trackSearchProperties == WFindOnWebMenu::TrackSearchProperties::Title ||
+                trackSearchProperties == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+            return kSearchUrlSoundCloudTitle;
+        } else if (trackSearchProperties == WFindOnWebMenu::TrackSearchProperties::Artist) {
+            return kSearchUrlSoundCloudArtist;
+        } else {
+            return kSearchUrlSoundCloudAlbum;
+        }
+    default:
+        return kSearchUrlDefault;
+    }
+}
+} // namespace
+
+WFindOnWebMenu::WFindOnWebMenu(QWidget* parent)
+        : QMenu(tr("Find on Web"), parent) {
+}
+
+void WFindOnWebMenu::addSubmenusForServices(const Track& track) {
+    WFindOnWebMenu::populateFromTrackProperties(
+            track,
+            QString(tr("SoundCloud")),
+            Service::SoundCloud);
+    WFindOnWebMenu::populateFromTrackProperties(
+            track, QString(tr("LastFm")), Service::LastFm);
+    WFindOnWebMenu::populateFromTrackProperties(
+            track, QString(tr("Discogs")), Service::Discogs);
```

Comment:
I don't think these have to / should be translatable.

Reply:
Thank you for the feedback. 

I will change them to QStringLiteral.